pkg/cmd/experimental/bundlesecret/bundle-secret.go

```diff
@@ -0,0 +1,99 @@
+package bundlesecret
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/spf13/cobra"
+	"github.com/golang/glog"
+)
+
+func NewCmdBundleSecret(f *clientcmd.Factory, parentName, name string, out io.Writer) *cobra.Command {
+	var sourceList util.StringList
+	cmd := &cobra.Command{
+		Use:   "bundle-secret NAME -f SOURCENAME",
+		Short: "bundle data from a source list to load kubernetes secret",
+		Long:  "bundle data from a source list to load kubernetes secret",
+		Run: func(cmd *cobra.Command, args []string) {
+
+			if len(args) == 0 {
+				glog.Fatalf("Must specify NAME as an argument")
+			}
+			objName := args[0]
+
+			if len(sourceList) == 0 {
+				glog.Fatalf("Must specify files and/or directories to gather data from.")
+			}
+
+			secretData := make(map[string][]byte)
+
+			for _, sourceItem := range sourceList {
+				sourceItem = strings.TrimSuffix(sourceItem, "/")
+				info, err := os.Stat(sourceItem)
+				checkErr(err)
+
+				if info.IsDir() {
+					fileList, err := ioutil.ReadDir(sourceItem)
+					checkErr(err)
+
+					for _, item := range fileList {
+						if !item.IsDir() {
+							file := fmt.Sprint(sourceItem, "/", item.Name())
+							err := readFiles(file, item.Name(), secretData, out)
+							checkErr(err)
+				        }
+				    } 
+				} else {
+					err := readFiles(sourceItem, info.Name(), secretData, out)
+					checkErr(err)
+				}
+			}
+
+			secretObj := kapi.Secret{
+				ObjectMeta: kapi.ObjectMeta{Name: objName},
+				Data:       secretData,
+			}
+
+			setDefaultPrinter(cmd)
+			f.Factory.PrintObject(cmd, &secretObj, out)
+		},
+	}
+
+	cmd.Flags().VarP(&sourceList, "source", "f", "List of filenames, directories to use as source of Kubernetes Secret.Data")
+	kcmdutil.AddPrinterFlags(cmd)
+	return cmd
+}
+
+func checkErr(err error) {
+	if err != nil {
+		glog.Fatal(err)
+	}
+}
+
+func setDefaultPrinter(c *cobra.Command) {
+	flag := c.Flags().Lookup("output")
+	if len(flag.Value.String()) == 0 {
+		flag.Value.Set("json")
+	}
+}
+
+func readFiles(filePath, fileName string, dataMap map[string][]byte, out io.Writer) error {
```

Comment:
change to:

```
func readFile(file os.FileInfo, dataMap map[string][]byte) error
```

You do not use ```out``` so seems pointless to pass now.
You have a fileInfo object from the calling class, so the detail of working with the path to read the file seems better situated in this function.